sql-plugin/src/main/330+/scala/org/apache/spark/sql/rapids/shims/intervalExpressions.scala

```diff
@@ -0,0 +1,497 @@
+        }
+      case _ => throw new IllegalArgumentException(
+        s"Not support num type $numType in GpuMultiplyYMInterval")
+    }
+  }
+
+  override def toString: String = s"$interval * $num"
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(YearMonthIntervalType, NumericType)
+
+  override def dataType: DataType = YearMonthIntervalType()
+}
+
+/**
+ * Multiply a day-time interval by a numeric
+ * day-time interval * number(byte, short, int, long, float, double)
+ * Note not support day-time interval * decimal
+ * Day-time interval's interval type is long, the value of long is the total microseconds
+ * Rewrite from Spark code:
+ * https://github.com/apache/spark/blob/v3.2.1/sql/catalyst/src/main/scala/
+ * org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala#L558
+ */
+case class GpuMultiplyDTInterval(
+    interval: Expression,
+    num: Expression)
+    extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  override def left: Expression = interval
+
+  override def right: Expression = num
+
+  override def doColumnar(interval: GpuColumnVector, numScalar: GpuScalar): ColumnVector = {
+    doColumnarImp(interval.getBase, numScalar.getBase, num.dataType)
+  }
+
+  override def doColumnar(interval: GpuColumnVector, num: GpuColumnVector): ColumnVector = {
+    doColumnarImp(interval.getBase, num.getBase, num.dataType)
+  }
+
+  override def doColumnar(intervalScalar: GpuScalar, num: GpuColumnVector): ColumnVector = {
+    doColumnarImp(intervalScalar.getBase, num.getBase, num.dataType)
+  }
+
+  override def doColumnar(numRows: Int, intervalScalar: GpuScalar,
+      numScalar: GpuScalar): ColumnVector = {
+    withResource(GpuColumnVector.from(intervalScalar, numRows, interval.dataType)) { expandedLhs =>
+      doColumnar(expandedLhs, numScalar)
+    }
+  }
+
+  private def doColumnarImp(interval: BinaryOperable, numOperable: BinaryOperable,
+      numType: DataType): ColumnVector = {
+    numType match {
+      case ByteType | ShortType | IntegerType => // num is byte, short or int
+        // compute interval.asLong * num
+        // interval and num are both in the range: [Int.MinValue, Int.MaxValue],
+        // so long fits the result and no need to check the overflow
+        val longResultCv: ColumnVector = IntervalUtils.mulIntervalAndNum(interval, numOperable)
+
+        withResource(longResultCv) { longResult =>
+          // check overflow
+          IntervalUtils.checkMultiplyOverflow(interval, numOperable, longResult)
+
+          longResult.incRefCount()
+        }
+
+      case LongType => // num is long
+        // compute interval(long) * num
+        val longResultCv: ColumnVector = interval.mul(numOperable)
+
+        withResource(longResultCv) { longResult =>
+          // check overflow
+          IntervalUtils.checkMultiplyOverflow(interval, numOperable, longResult)
+
+          longResult.incRefCount()
+        }
+      case _: FloatType => // num is float
```

Comment:
Did you try
```
case FloatType | DoubleType => interval.mul(numOperable, DType.FLOAT64)
```
instead of separating the float and double ?

Reply:
done